src/batched/dense/impl/KokkosBatched_Gemv_TeamVector_Impl.hpp

```diff
@@ -68,30 +72,34 @@ struct TeamVectorGemv<MemberType, Trans::Transpose, Algo::Gemv::Unblocked> {
   KOKKOS_INLINE_FUNCTION static int invoke(
       const MemberType &member, const ScalarType alpha, const AViewType &A,
       const xViewType &x, const ScalarType beta, const yViewType &y) {
-    if (AViewType::Rank == 2)
-      return TeamVectorGemvInternal<Algo::Gemv::Unblocked>::invoke(
-          member, A.extent(1), A.extent(0), alpha, A.data(), A.stride_1(),
-          A.stride_0(), x.data(), x.stride_0(), beta, y.data(), y.stride_0());
-    else
-      return TeamVectorGemvInternal<Algo::Gemv::Unblocked>::template invoke<
-          MemberType, ScalarType, typename AViewType::array_layout,
-          typename AViewType::non_const_value_type>(
-          member, A.extent(0), A.extent(2), A.extent(1), alpha, A.data(),
-          A.stride_0(), A.stride_2(), A.stride_1(), x.data(), x.stride_0(),
-          x.stride_1(), beta, y.data(), y.stride_0(), y.stride_1());
+    static_assert(AViewType::Rank == 3,
+                  "Batched TeamVectorGemv requires rank-3 A matrix (use "
+                  "KokkosBlas::TeamVectorGemv for regular rank-2 matrix)");
+    return TeamVectorGemvInternal<Algo::Gemv::Unblocked>::template invoke<
+        MemberType, ScalarType, typename AViewType::array_layout,
+        typename AViewType::non_const_value_type>(
+        member, A.extent(0), A.extent(2), A.extent(1), alpha, A.data(),
+        A.stride_0(), A.stride_2(), A.stride_1(), x.data(), x.stride_0(),
+        x.stride_1(), beta, y.data(), y.stride_0(), y.stride_1());
```

Comment:
@lucbv @kliegeois 
Batched GEMV routines (with rank-3 `A` matrix and rank-2 `x`, `y` vectors) stay in `KokkosBatched`, while non-batched routines are moved to `KokkosBlas` (naming without changes).